### PR TITLE
LTD-4616-add-is-query-closed-property 

### PR DESCRIPTION
**Pull request by depsiatwal**
https://uktrade.atlassian.net/browse/LTD-4616

A new property is added to ECJUQuery object currently when a query is being tested for closure it's inconsistent 
as the code checked response is null , response_at is null. This property decouples these checks so we can test within the model. 

The are further designs coming into scope where an exporter can close a query without having a response. So this will allow the model to support this change.

### CODE REVIEW

File: api/cases/serializers.py [hunk 1]
```diff
@@ -241,6 +242,9 @@ def get_raised_by_user(self, instance):
     def get_responded_by_user(self, instance):
         return self._user_name(instance.responded_by_user)
 
+    def get_is_query_closed(self, instance):
+        return self._user_name(instance.is_query_closed)
```
Comment (kevincarrogan) [inline api/cases/serializers.py:246]: I can see that this is a copy-paste of some other code but calling _user_name is a super confusing thing to do. I don’t really understand how this user name thing is a function.

Comment (depsiatwal) [reply api/cases/serializers.py]: not sure what I was thinking here. 
was trying to copy cat the code but just need a normal boolean field.

File: api/cases/models.py [hunk 2]
```diff
@@ -612,6 +612,10 @@ class EcjuQuery(TimestampableModel):
         choices=ECJUQueryType.choices, max_length=50, default=ECJUQueryType.ECJU, null=False, blank=False
     )
 
+    @property
+    def is_query_closed(self):
+        return self.responded_at is not None
+
```
Comment (hnryjmes) [inline api/cases/models.py:618]: nice, couldn't have done it better myself

**Review by hnryjmes — APPROVED** [review]
looking good

File: api/cases/models.py (hunk 2 above)
Comment (depsiatwal) [reply api/cases/models.py:618]: loll you did it was copied from your PR

**Review by kevincarrogan — APPROVED** [review]

**Comment by depsiatwal** [comment]
This PR has been changed to use a new library to allow filtering and queries based on properties.  This has now been broken down into two parts

This PR - add the library and a queryable property  

Part 2 to follow : this will allow filtering and update the API to use the new filters.

**Pull request merged**